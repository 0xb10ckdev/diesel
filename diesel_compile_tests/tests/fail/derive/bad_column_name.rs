#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
        #[sql_name = "type"]
        tpe -> Text,
    }
}

#[derive(Queryable)]
#[diesel(table_name = users)]
struct User1 {
    #[diesel(column_name)]
    name: String,
}

#[derive(Queryable)]
#[diesel(table_name = users)]
struct User2 {
    #[diesel(column_name(another))]
    name: String,
}

#[derive(Queryable)]
#[diesel(table_name = users)]
struct User3 {
    #[diesel(column_name = true)]
    name: String,
}


#[derive(Insertable)]
#[diesel(table_name = users)]
struct User4 {
    #[diesel(column_name = "type")]
    ty: String,
}


#[derive(AsChangeset)]
#[diesel(table_name = users)]
struct User5 {
    #[diesel(column_name = "type")]
    ty: String,
}

fn main() {}
