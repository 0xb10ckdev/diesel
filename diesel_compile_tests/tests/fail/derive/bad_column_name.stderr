error: unexpected end of input, expected `=`
  --> tests/fail/derive/bad_column_name.rs:14:25
   |
14 |     #[diesel(column_name)]
   |                         ^
   |
   = help: The correct format looks like `#[diesel(column_name = foo)]`

error: expected `=`
  --> tests/fail/derive/bad_column_name.rs:21:25
   |
21 |     #[diesel(column_name(another))]
   |                         ^

error: expected string literal
  --> tests/fail/derive/bad_column_name.rs:28:28
   |
28 |     #[diesel(column_name = true)]
   |                            ^^^^
